### PR TITLE
SW-4713 Add provider, context, routing, and 'single' view for participant project

### PR DESCRIPTION
**Pull request by nickgraz**

### CODE REVIEW

**Comment by nickgraz** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2377?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2377** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2377?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2375** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2375?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nickgraz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/hooks/navigation/useParticipantProjectsNav.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { useCallback } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { APP_PATHS } from 'src/constants';
+
+export default function useParticipantProjectsNav(projectId?: number) {
```
Comment (karthikbtf) [inline src/hooks/navigation/useParticipantProjectsNav.tsx:6]: The other hook was `useNavigateToParticipants` we could rename this and file to `useNavigateToParticipantProjects` or rename the other one. I have no preference except to have some convention.

File: src/hooks/navigation/useParticipantProjectsNav.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  const history = useHistory();
+
+  return {
```
Comment (karthikbtf) [inline src/hooks/navigation/useParticipantProjectsNav.tsx:9]: Nice! Should we have just one navigate hook with all the options exposed? 
`src/hooks/useNavigateTo` and merge this hook with the other.. thoughts?

File: src/redux/features/participantProjects/participantProjectsAsyncThunks.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ export const requestGetParticipantProject = createAsyncThunk(
   'participantProjects/get-one',
   async (participantProjectId: number, { rejectWithValue }) => {
     const response: Response2<ParticipantProjectData> = await ParticipantProjectService.get(participantProjectId);
-
+    console.log(response);
```
Comment (karthikbtf) [inline src/redux/features/participantProjects/participantProjectsAsyncThunks.ts:11]: rm?

File: src/scenes/AcceleratorRouter/ParticipantProjects/SingleView.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    <Grid container spacing={theme.spacing(1)}>
+      <Grid item xs={10}>
+        <Page
+          title={`${project?.organizationName || ''} / ${project?.name || ''}`}
+          crumbs={crumbs}
+          hierarchicalCrumbs={false}
+          rightComponent={rightComponent}
+        >
+          {status === 'pending' && <BusySpinner />}
+          <Card style={{ width: '100%' }} />
```
Comment (karthikbtf) [inline src/scenes/AcceleratorRouter/ParticipantProjects/SingleView.tsx:45]: flex?

**Review by karthikbtf — APPROVED** [review]
Nice and clean. One question to get your thoughts on a single nav hook file.

File: src/hooks/navigation/useParticipantProjectsNav.tsx (hunk 2 above)
Comment (nickgraz) [reply src/hooks/navigation/useParticipantProjectsNav.tsx:9]: I like it, I'll make that change in here

**Review by karthikbtf — APPROVED** [review]
Awesome, thanks!!

**Comment by nickgraz** [comment]
### Merge activity

* **Mar 21, 2:24 PM EDT**: @nickgraz started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2377).

**Pull request merged**